### DISCUSSION TITLE
Bug

### DISCUSSION THREAD

**Discussion by AlainZuur** [JenniNiku/gllvm#165] [General]
Hello,

This works fine when using the CRAN version, but with the development version it gives me an error:

Ysim <- simulate(M2, nsim = NSim, conditional = FALSE)

M2 is a Poisson gllvm model with 2 LVs and a couple of covariates.

Same happens with the spider data on your website. And with nsim=1 it worked ok (in the development version).

Switching back to the CRAN version solves it for the moment.

I can re-install the development version and copy-paste the error...if that helps.

Alain

Sometimes using gllvm::simulate()  or simulate.gllvm()  helps...but not in this case.

**Comment by BertvanderVeen** [comment]
Thanks. Can you see if my commit fixed your error? Also, feel free to post bugs under "issues".

**Reply by AlainZuur** [reply]
Sorry Bert. I'm not a github hero.

Your commit...does that mean that if I rerun:

devtools::install_github("JenniNiku/gllvm")

then that should solve the problem? I don't think it did. For the spider data:

predict(model = MGLM, type = "response", level = 1)  
Error in predict.gllvm(model = MGLM, type = "response", level = 1) : 
  argument "object" is missing, with no default


simulate(MGLM, nsim = 2, conditional = FALSE)
Error in predict.gllvm(object, newLV = lvsNew, type = "response") : 
  Number of rows in 'newLV' must be the same as the number of rows in the data used for model fitting, if new X values for the corresponding units (in 'newLV') are not given.



 Ysim <- simulate.gllvm(M2, nsim = NSim, conditional = FALSE)
Error in predict.gllvm(object, newX = Xnew, newLV = lvsNew, type = "response") : 
  object 'model' not found



Coming from this:
library(gllvm)

data(spider)
Y <- spider$abund
X <- spider$x

#And scale the predictors
X <- scale(X)

MGLM <- gllvm(Y, X = X, 
                          formula = ~ bare.sand + fallen.leaves,
              family = "poisson", num.lv = 2)

predict(model = MGLM, type = "response", level = 1)  
simulate(MGLM, nsim = 2, conditional = FALSE)


Alain

**Reply by BertvanderVeen** [reply]
No worries Alain! I will still respond if you don't post under "issues". Apologies, this latest error is due to a typo of mine in the code while making the commit. That is what I get for trying to keep up with github questions while also preparing for a workshop on GLMs I suppose.. Please reinstall from the development branch one more time.

**Reply by AlainZuur** [reply]
Okodiki.

Alain

**Comment by AlainZuur** [comment]
Yes...your latest version works....so predict and simulate now work in the development version.

The development version...does it install things that the CRAN version then uses later? Just before this message, the CRAN version also started to give errors with predict (which it didn't do before). Anyway...all is fine now. 

Good luck with your course. There is nothing more time-consuming than making course material! But having 20 people going through your material is the best refereeing system.

Alain

PS..I am on R version 4.3.2 ....using a Mac.

**Reply by AlainZuur** [reply]
Thanks for your fast response! Appreciated.

Alain

**Reply by BertvanderVeen** [reply]
> Yes...your latest version works....so predict and simulate now work in the development version.
> 
> The development version...does it install things that the CRAN version then uses later? Just before this message, the CRAN version also started to give errors with predict (which it didn't do before). Anyway...all is fine now. 
> 
> Good luck with your course. There is nothing more time-consuming than making course material! But having 20 people going through your material is the best refereeing system.
> 
> Alain
> 
> PS..I am on R version 4.3.2 ....using a Mac.
> 

It shouldn't. As long as you don't try to install or load both packages at the same time, that is. Did you use a clean environment for both attempts?

**Comment by AlainZuur** [comment]
Yes..restart of RStudio...nothing saved.

I think the development version did touch the CRAN version. I will try a clean install on an RStudio on posit (posit.... by the way, may be very useful for your GLM course)

Anyway...everything is working. Have a nice weekend.

Alain
